### ISSUE TITLE
unused_closure_parameter false positive with SwiftUI Binding

### ISSUE THREAD

**Issue by freak4pc** [realm/SwiftLint#3790]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

The following code should not trigger a warning. It is legal to use `$binding` on a closure argument to get its `wrappedValue` in `binding`:
<img width="578" alt="image" src="https://user-images.githubusercontent.com/605076/147387508-ea670d65-8ff6-471e-b5fa-dce3b9bc7693.png">

This code is equivalent to manually writing:
<img width="538" alt="image" src="https://user-images.githubusercontent.com/605076/147387513-20e4b9a6-b397-40ad-8552-bd663e844bb1.png">

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 0.45.1
* Installation method used (Homebrew, CocoaPods, building from source, etc)? Portable binary
* Paste your configuration file: 

```yml
excluded:
  - Tuist

line_length: 200
```
